lineEnding=KEEP breaks JSON validation on Windows

[test.zip](https://github.com/revelc/formatter-maven-plugin/files/14105204/test.zip)
**Describe the bug**
A clear and concise description of what the bug is.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 2.23.0
 - OS: Window 10 x64
- Maven: 3.9.4
- Java 17 (OpenJDK build 17+35-2724)

**To Reproduce**

When I run `mvn formatter:validate` on the attached project I get the following error:
```
[INFO] --- formatter:2.23.0:validate (default-cli) @ test ---
[WARNING] 
com.fasterxml.jackson.databind.JsonMappingException: Cannot invoke "String.length()" because "text" is null (through reference chain: java.util.LinkedHashMap["properties"])
    at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath (JsonMappingException.java:402)
    at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath (JsonMappingException.java:361)
    at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow (StdSerializer.java:323)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields (MapSerializer.java:811)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo (MapSerializer.java:764)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize (MapSerializer.java:720)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize (MapSerializer.java:35)
    at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize (DefaultSerializerProvider.java:479)
    at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue (DefaultSerializerProvider.java:318)
    at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize (ObjectWriter.java:1572)
    at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose (ObjectWriter.java:1273)
    at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString (ObjectWriter.java:1140)
    at net.revelc.code.formatter.json.JsonFormatter.doFormat (JsonFormatter.java:84)
    at net.revelc.code.formatter.AbstractCacheableFormatter.formatFile (AbstractCacheableFormatter.java:83)
    at net.revelc.code.formatter.FormatterMojo.doFormatFile (FormatterMojo.java:776)
    at net.revelc.code.formatter.ValidateMojo.doFormatFile (ValidateMojo.java:74)
    at net.revelc.code.formatter.FormatterMojo.formatFile (FormatterMojo.java:694)
    at net.revelc.code.formatter.FormatterMojo.execute (FormatterMojo.java:469)
    at net.revelc.code.formatter.ValidateMojo.execute (ValidateMojo.java:67)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
Caused by: java.lang.NullPointerException: Cannot invoke "String.length()" because "text" is null
    at com.fasterxml.jackson.core.json.WriterBasedJsonGenerator.writeRaw (WriterBasedJsonGenerator.java:554)
    at com.fasterxml.jackson.core.util.DefaultIndenter.writeIndentation (DefaultIndenter.java:90)
    at com.fasterxml.jackson.core.util.DefaultPrettyPrinter.beforeObjectEntries (DefaultPrettyPrinter.java:303)
    at com.fasterxml.jackson.core.json.WriterBasedJsonGenerator._writePPFieldName (WriterBasedJsonGenerator.java:371)
    at com.fasterxml.jackson.core.json.WriterBasedJsonGenerator._writeFieldName (WriterBasedJsonGenerator.java:172)
    at com.fasterxml.jackson.core.json.WriterBasedJsonGenerator.writeFieldName (WriterBasedJsonGenerator.java:155)
    at com.fasterxml.jackson.databind.ser.std.StdKeySerializers$StringKeySerializer.serialize (StdKeySerializers.java:302)
    at com.fasterxml.jackson.databind.ser.std.StdKeySerializers$Dynamic.serialize (StdKeySerializers.java:264)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeFields (MapSerializer.java:797)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serializeWithoutTypeInfo (MapSerializer.java:764)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize (MapSerializer.java:720)
    at com.fasterxml.jackson.databind.ser.std.MapSerializer.serialize (MapSerializer.java:35)
    at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize (DefaultSerializerProvider.java:479)
    at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue (DefaultSerializerProvider.java:318)
    at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize (ObjectWriter.java:1572)
    at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose (ObjectWriter.java:1273)
    at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString (ObjectWriter.java:1140)
    at net.revelc.code.formatter.json.JsonFormatter.doFormat (JsonFormatter.java:84)
    at net.revelc.code.formatter.AbstractCacheableFormatter.formatFile (AbstractCacheableFormatter.java:83)
    at net.revelc.code.formatter.FormatterMojo.doFormatFile (FormatterMojo.java:776)
    at net.revelc.code.formatter.ValidateMojo.doFormatFile (ValidateMojo.java:74)
    at net.revelc.code.formatter.FormatterMojo.formatFile (FormatterMojo.java:694)
    at net.revelc.code.formatter.FormatterMojo.execute (FormatterMojo.java:469)
    at net.revelc.code.formatter.ValidateMojo.execute (ValidateMojo.java:67)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.23.0:validate (default-cli) on project test: Error formatting 'D:\test\src\main\resources\META-INF\additional-spring-configuration-metadata.json'  -> [Help 1]
```
When `<lineEnding>KEEP</lineEnding>` is removed, plugin functions correctly:
```
[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.23.0:validate (default-cli) on project test: File 'D:\test\src\main\resources\META-INF\additional-spring-configuration-metadata.json' has not been previously formatted. Please format file (for example by invoking `mvn net.revelc.code.formatter:formatter-maven-plugin:2.23.0:format`) and commit before running validation! -> [Help 1]
```

--- a/src/main/java/net/revelc/code/formatter/FormatterMojo.java
+++ b/src/main/java/net/revelc/code/formatter/FormatterMojo.java
@@ -929,13 +929,17 @@ public class FormatterMojo extends AbstractMojo implements ConfigurationSource {
         // Xml Setup
         if (this.configXmlFile != null) {
             final var xmlFormattingOptions = this.getOptionsFromPropertiesFile(this.configXmlFile);
-            xmlFormattingOptions.put("lineending", this.lineEnding.getChars());
+            if (this.lineEnding != LineEnding.KEEP) {
+                xmlFormattingOptions.put("lineending", this.lineEnding.getChars());
+            }
             this.xmlFormatter.init(xmlFormattingOptions, this);
         }
         // Json Setup
         if (this.configJsonFile != null) {
             final var jsonFormattingOptions = this.getOptionsFromPropertiesFile(this.configJsonFile);
-            jsonFormattingOptions.put("lineending", this.lineEnding.getChars());
+            if (this.lineEnding != LineEnding.KEEP) {
+                jsonFormattingOptions.put("lineending", this.lineEnding.getChars());
+            }
             this.jsonFormatter.init(jsonFormattingOptions, this);
         }
         // Css Setup

--- a/src/main/java/net/revelc/code/formatter/json/JsonFormatter.java
+++ b/src/main/java/net/revelc/code/formatter/json/JsonFormatter.java
@@ -18,6 +18,7 @@ import java.io.IOException;
 import java.io.StringWriter;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.regex.Pattern;
 
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.util.DefaultIndenter;
@@ -38,6 +39,8 @@ public class JsonFormatter extends AbstractCacheableFormatter implements Formatt
 
     /** The formatter. */
     private ObjectMapper formatter;
+
+    private static final Pattern ANY_EOL = Pattern.compile("\\R");
 
     @Override
     public void init(final Map<String, String> options, final ConfigurationSource cfg) {
@@ -82,7 +85,7 @@ public class JsonFormatter extends AbstractCacheableFormatter implements Formatt
             final Iterator<Object> jsonObjectIterator = this.formatter.readValues(jsonParser, Object.class);
             while (jsonObjectIterator.hasNext()) {
                 String jsonString = this.formatter.writer().writeValueAsString(jsonObjectIterator.next());
-                stringWriter.write(jsonString.strip().replaceAll("\\R", ending.getChars()));
+                stringWriter.write(ANY_EOL.matcher(jsonString.strip()).replaceAll(ending.getChars()));
                 stringWriter.write(ending.getChars());
             }
             String formattedCode = stringWriter.toString();

--- a/src/main/java/net/revelc/code/formatter/json/JsonFormatter.java
+++ b/src/main/java/net/revelc/code/formatter/json/JsonFormatter.java
@@ -79,11 +79,10 @@ public class JsonFormatter extends AbstractCacheableFormatter implements Formatt
     protected String doFormat(final String code, final LineEnding ending) throws IOException {
         try (StringWriter stringWriter = new StringWriter()) {
             JsonParser jsonParser = this.formatter.createParser(code);
-            // note: line ending set in init for this usecase
             final Iterator<Object> jsonObjectIterator = this.formatter.readValues(jsonParser, Object.class);
             while (jsonObjectIterator.hasNext()) {
                 String jsonString = this.formatter.writer().writeValueAsString(jsonObjectIterator.next());
-                stringWriter.write(jsonString);
+                stringWriter.write(jsonString.strip().replaceAll("\\R", ending.getChars()));
                 stringWriter.write(ending.getChars());
             }
             String formattedCode = stringWriter.toString();

